writeToStream.js

```diff
@@ -0,0 +1,579 @@
+
+'use strict';
+
+var protocol = require('./constants')
+  , empty = new Buffer(0)
+  , zeroBuf = new Buffer([0])
+  , numCache = require('./numbers')
+
+function generate(packet, stream) {
+
```

Comment:
Also note that you can't collide calls. Meaning, if a user called `cork`/`uncork` inside your own it won't cause any problems with yours.